.github/workflows/daily.yaml

```diff
@@ -87,29 +87,45 @@ jobs:
               echo "Attempting to update sources of ${chartname}..."
               echo "Using go-yq verion: <$(go-yq -V)>"
               # Get all sources (except truecharts)
-              curr_sources=$(go-yq '.sources[] | select(. != "https://github.com/truecharts*" and . != "https://ghcr*" and . != "https://hub.docker*" and . != "https://quay*" and . != "https://github.com/truecharts/containers*")' "${chart}/Chart.yaml")
+              curr_sources=$(go-yq '.sources[] | select(. != "https://github.com/truecharts*" and . != "https://ghcr*" and . != "https://hub.docker*" and . != "https://quay*" and . != "https://github.com/truecharts/containers*" and test("^https?://"))' "${chart}/Chart.yaml")
               # Empty sources list in-place
               go-yq -i 'del(.sources.[])' "${chart}/Chart.yaml"
               # Add truechart source
               tcsource="https://github.com/truecharts/charts/tree/master/charts/$train/$chartname" go-yq -i '.sources += env(tcsource)' "${chart}/Chart.yaml"
               container=$(cat website/docs/charts/description_list.md | grep "\[${chartname}\]" | cut -f3 -d '|' | grep -v 'Not Found' || echo "")
               if [ ! -z "$container" ]; then
-                prefix="https://hub.docker.com/"
-                if echo ${container} | grep 'ghcr'; then
-                prefix="https://ghcr.com/"
-                elif echo ${container} | grep 'quay'; then
-                prefix="https://quay.com/"
-                elif echo ${container} | grep 'tccr'; then
-                prefix="https://github.com/truecharts/containers/tree/master/mirror/"
-                fi
+                case "$container" in
+                  *ghcr*|*quay*)
+                    # Both ghcr.io and quay.io have nice redirection.
+                    prefix="https://"
+                    ;;
+                  *lscr*)
+                    # Images hosted on lscr.io seem to all be available on Docker Hub under the linuxserver user.
+                    prefix="https://hub.docker.com/r/linuxserver/"
+                    # Get the image name without any pathing.
+                    container="${container##*/}"
+                    ;;
+                  *tccr*)
```

Comment:
I wasn't really confident in this matching, it seemed very susceptible to collisions, like if I made an `opello/ghcr` image and pushed it to Docker Hub, I think a bad link would be generated.

Reply:
I ran through all of the substrings and didn't find a problem in the generated output.  I still think it's worth thinking about a little though.